test/mscclpp-test/allgather_test.cu

```diff
@@ -288,6 +288,49 @@ __global__ void allgather4(int rank, int worldSize, int nRanksPerNode, size_t ne
                    nBlocksForLocalAllGather);
 }
 
+__global__ void allgather5(int rank, int worldSize, int nRanksPerNode, size_t nelemsPerGPU) {
+  const int tid = threadIdx.x + blockIdx.x * blockDim.x;
+  const int lid = tid % WARP_SIZE;
+  const int wid = tid / WARP_SIZE;
+  const int nWarp = blockDim.x * gridDim.x / WARP_SIZE;
+  const int nPeer = nRanksPerNode - 1;
+  const int chanOffset = nPeer * blockIdx.x;
+  auto smChans = constSmChans + chanOffset;
+
+  if (wid < nPeer) {
+    smChans[wid].signal();
+    smChans[wid].wait();
+  }
```

Comment:
Do we need all threads in a warp to call signal/wait? Or just need one thread in a warp
```cpp
if (wid < nPeer and lid == 0)
```

Reply:
@Binyang2014 Thank you for finding this! Fixed in [cc9c150](https://github.com/microsoft/mscclpp/pull/248/commits/cc9c1505207848efd1d7d68dcaddac48387142a7).